sgtl/graph.py

```diff
@@ -330,6 +330,39 @@ def tensor_product(self, other_graph):
         return Graph(scipy.sparse.kron(self.adjacency_matrix(), other_graph.adjacency_matrix()))
 
 
+def hypercube_graph(d: int) -> Graph:
+    """
+    Construct the hypercube unweighted graph on :math:`2**d` vertices.
+    :param d: Degree of the hypercube graph.
+    :return: The hypercube graph on :math:`2**d` vertices, as a `Graph` object.
+    :raises ValueError: if the degree is not a positive integer.
+    :Example:
+       >>> import sgtl.graph
+       >>> graph = sgtl.graph.hypercube_graph(3)
+       >>> graph.adjacency_matrix().toarray()
+       array([[0., 1., 1., 0., 1., 0., 0., 0.],
+              [1., 0., 0., 1., 0., 1., 0., 0.],
+              [1., 0., 0., 1., 0., 0., 1., 0.],
+              [0., 1., 1., 0., 0., 0., 0., 1.],
+              [1., 0., 0., 0., 0., 1., 1., 0.],
+              [0., 1., 0., 0., 1., 0., 0., 1.],
+              [0., 0., 1., 0., 1., 0., 0., 1.],
+              [0., 0., 0., 1., 0., 1., 1., 0.]])
+    """
+    if d <= 0:
```

Comment:
```suggestion
# pylint: disable=invalid-name
    if d <= 0:
```

This method causes the checks on the code to fail since the variable name `d` is not considered good practice. I'm happy to ignore this error, but we need to explicitly override the linter, which can be done with the above line.

For another example, see the `ssbm` method in the `random` module.